### PR TITLE
SPORestrictedAccessForSitesInsights cmdlet documentation

### PR DESCRIPTION
**Pull request by ishwarit**
[PubOps PR review](https://github.com/MicrosoftDocs/OfficeDocs-SharePoint-PowerShell/pull/715#issuecomment-2436507111)
***

Cmdlet documentation for SPORestrictedAccessForSitesInsights Start and Get cmds.

### CODE REVIEW

**Comment by ishwarit** [comment]
@samkabue Please review and approve the documentation PR.

**Review by AdwoodMicrosoft — APPROVED** [review]

File: sharepoint/sharepoint-ps/sharepoint-online/Start-SPORestrictedAccessForSitesInsights [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+Example 1 generates the Restricted access control policy report which contains insights about the list of sites protected with the policy.
+
+### -----------------------EXAMPLE 2-----------------------------
+
+```powershell
+Start-SPORestrictedAccessForSitesInsights -ActionsBlockedByPolicy
+```
+
+Example 2 generates the Restricted access control policy report which contains insights about access denails by the policy.
```
Comment (samkabue) [inline sharepoint/sharepoint-ps/sharepoint-online/Start-SPORestrictedAccessForSitesInsights:56]: >denails [](http://example.com/codeflow?start=101&length=7)

typo

**Comment by Stacyrch140** [comment]
Hi @ishwarit - 

**Following are blocking issues that need to be resolved before we can merge:**

sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights
sharepoint/sharepoint-ps/sharepoint-online/Start-SPORestrictedAccessForSitesInsights
- Add an approved extension to each of these files (see References for approved extensions). Files must be an approved file type for the repo.

***

**Reference**

[Criteria used for pull request reviews](https://review.docs.microsoft.com/en-us/office-authoring-guide/pr-criteria?branch=main)

**Blocking**
- (Blocking) The following are the only file types that are allowed in Magic repos:
  - Private repos: md, yml, ps1, xml, jpg, png, gif, svg, csv
  - Public repos: md, yml, jpg, png, gif, svg, csv
  
All other file types must be stored elsewhere, such as Microsoft Download Center. For more information about Download Center, see [Publish or update a file on the Microsoft Download Center](https://review.learn.microsoft.com/en-us/office-authoring-guide/publish-download-center).

***

After you resolve the blocking issues, remove the **Back to submitter** label and add the **Sign off** label again. 

If you need help resolving issues in this PR, post your question in the [Ask an Admin Teams Channel](https://teams.microsoft.com/l/channel/19%3aea9a561f603c4fbb9cf3931a6849eae4%40thread.skype/Ask%2520an%2520Admin?groupId=7079ffbf-1420-4ee2-9ddc-a2c351337320&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47).


Thanks, 
Stacy

File: sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights.md [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+ms.author: itambakhe
+ms.reviewer:
+manager:
+---
+
+# Get-SPORestrictedAccessForSitesInsights
+
+## SYNOPSIS
+
+This cmdlet enables the administrator to check status of all active and available reports about insights on sites protected and access denials by Restricted access control policy when no report ID is present and to view or download the report if report ID is present.
```
Comment (samkabue) [inline sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights.md:18]: this sentence is hard to read. Consider making it shorter -- maybe break it into two

File: sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights.md [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+
+## SYNTAX
+
+```powershell
+Get-SPORestrictedAccessForSitesInsights -RACProtectedSites [-ReportId <Guid>] [-Action <ActionType>] [-InsightsSummary <Boolean>]
+```
+
+```powershell
+Get-SPORestrictedAccessForSitesInsights -ActionsBlockedByPolicy [-ReportId <Guid>] [-Content <ContentType>] [-Action <ActionType>]
```
Comment (samkabue) [inline sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights.md:27]: The cmdlet implementation could be improved to use ParameterSets for clarity. Please work with Dev to do this as an improvement as a followup change.

File: sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights.md [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+| SiteName                     | The name of the SharePoint site.                              |
+| SiteType                     | The type of the SharePoint site.                              |
+| SiteURL                      | The URL of the SharePoint site.                               |
+| SiteSensitivity              | The sensitivity label of the SharePoint site.                 |
+| PrimaryAdmin                 | Primary admin of the SharePoint Site                          |
+| BlockDownloadPolicy          | Boolean status of Block download policy on the site.          |
+| ConditionalAccessPolicy      | Authentication context set for the Conditional access policy  |
+| RestrictedAccessControlGroups| Entra Groups configured during policy configuration           |
+
+If this cmdlet is executed with `-RACProtectedSites` `-ReportId` `-InsightsSummary` as parameter, count of sites protected with restricted access control compared to total number of sites will be displayed.
```
Comment (samkabue) [inline sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights.md:53]: >count [](http://example.com/codeflow?start=98&length=5)

"the count"

Comment (samkabue) [inline sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights.md:53]: >restricted [](http://example.com/codeflow?start=128&length=10)

what's the right casing? Some places this is "Restricted access control" and other cases it's "restricted access control"

File: sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights.md [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+If this cmdlet is executed with `-ActionsBlockedByPolicy` `-ReportId` `-Content TopSites` as parameter, top 100 sites with highest access denials by Restricted access control will be displayed with the following properties:
+
+| Property                     | Description                                                   |
+|:-----------------------------|:--------------------------------------------------------------|
+| SiteName                     | The name of the SharePoint site.                              |
+| SiteURL                      | The URL of the SharePoint site.                               |
+| SiteSensitivity              | The sensitivity label of the SharePoint site.                 |
+| SiteType                     | The type of the SharePoint site.                              |
+| AccessDenialsCount           | Total number of access denials on the site.                   |
+| UniqueUsersBlocked           | Count of Unique users denied access on the site.              |
```
Comment (samkabue) [inline sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights.md:72]: >Unique [](http://example.com/codeflow?start=42&length=6)

please only use lowercase unless you're referencing a brand name. eg this should be "unique"

Please review this file and the other one

File: sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights.md [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+
+If this cmdlet is executed with `-ActionsBlockedByPolicy` `-ReportId` `-Content TopUsers` as parameter, top 10 users with highest access denials by Restricted access control will be displayed with the following properties:
+
+| Property                     | Description                                                   |
+|:-----------------------------|:--------------------------------------------------------------|
+| UserEmail                    | Email id of the blocked user.                                 |
```
Comment (samkabue) [inline sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights.md:78]: >id [](http://example.com/codeflow?start=39&length=2)

"address"

File: sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights.md [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+
+If this cmdlet is executed with `-RACProtectedSites` `-ReportId` as parameter, top 100 sites with highest page views that are protected by Restrcited access control will be displayed with the following properties:
+
+| Property                     | Description                                                   |
+|:-----------------------------|:--------------------------------------------------------------|
+| SiteName                     | The name of the SharePoint site.                              |
+| SiteType                     | The type of the SharePoint site.                              |
+| SiteURL                      | The URL of the SharePoint site.                               |
+| SiteSensitivity              | The sensitivity label of the SharePoint site.                 |
+| PrimaryAdmin                 | Primary admin of the SharePoint Site                          |
```
Comment (samkabue) [inline sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights.md:48]: >Site [](http://example.com/codeflow?start=65&length=4)

"site"

File: sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights.md [hunk 8]
```diff
@@ -0,0 +1,244 @@
...
+| BlockDownloadPolicy          | Boolean status of Block download policy on the site.          |
+| ConditionalAccessPolicy      | Authentication context set for the Conditional access policy  |
```
Comment (samkabue) [inline sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights.md:50]: >policy [](http://example.com/codeflow?start=87&length=6)

some sentences end with `.` while others don't -- please update so they're all consistent

**Review by samkabue — CHANGES_REQUESTED** [review]
:clock1:

File: sharepoint/sharepoint-ps/sharepoint-online/Start-SPORestrictedAccessForSitesInsights.md [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+ms.author: itambakhe
+ms.reviewer:
+manager:
+---
+ 
+# Start-SPORestrictedAccessForSitesInsights
+
+## SYNOPSIS
+
+This cmdlet enables administrator to trigger the build of a new Restricted Access Control insights report for the data from last 28 days.
```
Comment (samkabue) [inline sharepoint/sharepoint-ps/sharepoint-online/Start-SPORestrictedAccessForSitesInsights.md:18]: >Restricted Access Control [](http://example.com/codeflow?start=64&length=25)

is this the right casing? the other file does it differently

File: sharepoint/sharepoint-ps/sharepoint-online/Start-SPORestrictedAccessForSitesInsights.md [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+Start-SPORestrictedAccessForSitesInsights -RACProtectedSites [-Force <SwitchParameter>] 
+```
+
+```powershell
+Start-SPORestrictedAccessForSitesInsights -ActionsBlockedByPolicy [-Force <SwitchParameter>] 
+```
+
+## DESCRIPTION
+
+After this cmdlet is executed, the Restricted Access Control insights report generation request for the requested sub type gets queued in the pipeline and the below metadata is displayed with the following properties:
```
Comment (samkabue) [inline sharepoint/sharepoint-ps/sharepoint-online/Start-SPORestrictedAccessForSitesInsights.md:32]: >After this cmdlet is executed, the Restricted Access Control insights report generation request for the requested sub type gets queued in the pipeline and the below metadata is displayed with the following properties [](http://example.com/codeflow?start=0&length=216)

This sentence could be simplified. e.g., you could remove stuff like "gets queued in the pipeline" to just "the request is initiated" etc

File: sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights.md [hunk 11]
```diff
@@ -0,0 +1,189 @@
+---
+external help file: sharepointonline.xml
+Module Name: Microsoft.Online.SharePoint.PowerShell
+online version: https://learn.microsoft.com/powershell/module/sharepoint-online/get-sporestrictedaccessforsitesinsights
+applicable: SharePoint Online
+title: Get-SPORestrictedAccessForSitesInsights
+schema: 2.0.0
+author: itambakhe
+ms.author: itambakhe
```
Comment (samkabue) [inline sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights.md:9]: fix these values

File: sharepoint/sharepoint-ps/sharepoint-online/Start-SPORestrictedAccessForSitesInsights.md [hunk 12]
```diff
@@ -0,0 +1,113 @@
+---
+external help file: sharepointonline.xml
+Module Name: Microsoft.Online.SharePoint.PowerShell
+online version: https://learn.microsoft.com/powershell/module/sharepoint-online/start-sporestrictedaccessforsitesinsights
+applicable: SharePoint Online
+title: Start-SPORestrictedAccessForSitesInsights
+schema: 2.0.0
+author: itambakhe
+ms.author: itambakhe
```
Comment (samkabue) [inline sharepoint/sharepoint-ps/sharepoint-online/Start-SPORestrictedAccessForSitesInsights.md:9]: fix these values

File: sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights.md [hunk 13]
```diff
@@ -0,0 +1,189 @@
+---
+external help file: sharepointonline.xml
+Module Name: Microsoft.Online.SharePoint.PowerShell
+online version: https://learn.microsoft.com/powershell/module/sharepoint-online/get-sporestrictedaccessforsitesinsights
+applicable: SharePoint Online
+title: Get-SPORestrictedAccessForSitesInsights
+schema: 2.0.0
+author: ishwarit
+ms.author: itambakhe
```
Comment (samkabue) [inline sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights.md:9]: >itambakhe [](http://example.com/codeflow?start=11&length=9)

fix this value

File: sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights.md [hunk 14]
```diff
@@ -0,0 +1,189 @@
...
+
+If this cmdlet is executed with `-ActionsBlockedByPolicy` as parameter, it displays the status of all active and completed reports.
+  
+If this cmdlet is executed with `-ActionsBlockedByPolicy` `-ReportId` `-Content TopSites` as parameter, top 100 sites with the highest access denials by restricted access control will be displayed.
+
+If this cmdlet is executed with `-ActionsBlockedByPolicy` `-ReportId` `-Content TopUsers` as parameter, top 10 users with the highest access denials by restricted access control will be displayed.
+
+If this cmdlet is executed with `-ActionsBlockedByPolicy` `-ReportId` `-Content AllDenials` as parameter, most recent 100 access denials by restricted access control in the last 28 days will be displayed.
+
+If this cmdlet is executed with `-ActionsBlockedByPolicy` `-ReportId` `-Content SiteDistribution` as parameter, distribution of access denials by restricted access control on different site types will be displayed.
```
Comment (samkabue) [inline sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights.md:46]: the right place for these is in the parameters section.

this "DESCRIPTION" section is for a summary description of the cmdlet

File: sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights.md (hunk 2 above)
Comment (samkabue) [reply sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights.md]: suggestion from Copilot (obvious rewrite it your way so its clear)

```
This cmdlet allows the administrator to check the status of all active and available reports on insights related to site protection and access denials enforced by restricted access control.
```

Comment (ishwarit) [reply sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights.md]: Resolved

File: sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights.md (hunk 4 above)
Comment (ishwarit) [reply sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights.md]: Resolved

Comment (ishwarit) [reply sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights.md]: Made changes to follow the "restricted access control" casing in both the files.

File: sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights.md (hunk 5 above)
Comment (ishwarit) [reply sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights.md]: Resolved

File: sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights.md (hunk 6 above)
Comment (ishwarit) [reply sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights.md]: This content is removed from the file now.

File: sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights.md (hunk 7 above)
Comment (ishwarit) [reply sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights.md]: This content is removed from the file now.

File: sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights.md (hunk 8 above)
Comment (ishwarit) [reply sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights.md]: Made sure all the sentences end with ".". Also, this content is removed from the file now.

File: sharepoint/sharepoint-ps/sharepoint-online/Start-SPORestrictedAccessForSitesInsights.md (hunk 9 above)
Comment (ishwarit) [reply sharepoint/sharepoint-ps/sharepoint-online/Start-SPORestrictedAccessForSitesInsights.md]: Made changes to follow the "restricted access control" casing in both the files.

File: sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights.md (hunk 11 above)
Comment (ishwarit) [reply sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights.md:9]: Resolved.

File: sharepoint/sharepoint-ps/sharepoint-online/Start-SPORestrictedAccessForSitesInsights.md (hunk 12 above)
Comment (ishwarit) [reply sharepoint/sharepoint-ps/sharepoint-online/Start-SPORestrictedAccessForSitesInsights.md:9]: Resolved.

File: sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights.md (hunk 14 above)
Comment (ishwarit) [reply sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights.md:46]: Would like to keep it this way to help in showcasing the parameter combinations.

File: sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights.md (hunk 11 above)
Comment (samkabue) [reply sharepoint/sharepoint-ps/sharepoint-online/Get-SPORestrictedAccessForSitesInsights.md:9]: @ishwarit `ms.author` is not resolved

**Review by samkabue — APPROVED** [review]
:shipit:

**Pull request merged**